archive/imo/imo2011_q3.lean

```diff
@@ -0,0 +1,77 @@
+/-
+Copyright (c) 2021 David Renshaw. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Renshaw
+-/
+
+import data.real.basic
+
+/-!
+# IMO 2011 Q3
+
+Let f : ℝ → ℝ be a function that satisfies
+
+   f(x + y) ≤ y * f(x) + f(f(x))
+
+for all x and y. Prove that f(x) = 0 for all x ≤ 0.
+
+# Solution
+
+Direct translation of the solution found in https://www.imo-official.org/problems/IMO2011SL.pdf
+-/
+
+theorem imo2011_q3
+  (f : ℝ → ℝ) (hf : ∀ x y, f (x + y) ≤ y * f x + f (f x))
+  : ∀ x ≤ 0, f x = 0 :=
+begin
+  -- reparameterize
+  have hxt : (∀ x t, f t ≤ t * f x - x * f x + f (f x)),
```

Comment:
```suggestion
  have hxt : ∀ x t, f t ≤ t * f x - x * f x + f (f x),
```
You don't need those parentheses. Same for the other `have`s.

Reply:
Good point. I think I was nervous that the binder commas would have too high of precedence, but apparently that's not an issue. Updated.